src/networking.c

```diff
@@ -1606,6 +1608,8 @@ int _writeToClient(client *c, ssize_t *nwritten) {
         /* If the buffer was sent, set bufpos to zero to continue with
          * the remainder of the reply. */
         if ((int)c->sentlen == c->bufpos) {
+            if(c->buf_peak < (size_t)c->bufpos)
+                c->buf_peak = (size_t)c->bufpos;
```

Comment:
@ranshid Because in the normal case, in the old code, unless the reply we send to the client exceeds `buf_usable_size`, `bufpos` is not equal `buf_usable_size`, so `buf_peak` is not necessarily equal to `buf_usable_size`.

Reply:
@sundb - correct! that is exactly what this change is aiming to. we want the client buffer to be extended only when  the buffer was filled.